tests/unit/test_paginated_graphql_query.py

```diff
@@ -151,18 +164,112 @@ def test_given_a_query_and_a_number_of_elements_to_query_i_have_a_progress_bar(
 
 
 def test_given_a_query_without_a_number_of_elements_to_query_i_do_nothave_a_progress_bar(
-    graphql_client, capsys
+    graphql_client: GraphQLClient, http_client: HttpClient, capsys
 ):
     # given
     options = QueryOptions(disable_tqdm=False)
     number_of_elements_to_query = None
 
     # when
-    gen = PaginatedGraphQLQuery(graphql_client).execute_query_from_paginated_call(
+    gen = PaginatedGraphQLQuery(graphql_client, http_client).execute_query_from_paginated_call(
         QUERY, WHERE, options, "", number_of_elements_to_query
     )
     list(gen)
 
     # then
     captured = capsys.readouterr()
     assert captured.err == ""
+
+
+def test_given_a_query_returning_serialized_json_it_parses_json_fields(
+    http_client,
+):
+    # given
+    options = QueryOptions(disable_tqdm=False)
+    number_of_elements_to_query = 1
+    post_call_function = None
+    graphql_client = MagicMock(spec=GraphQLClient)
```

Comment:
we could make this a fixture too, and put it in conftest

Reply:
this is already a fixture in the conftest. 
But there is already another graphql_client fixture in the file to override the base on in the conftest so I need to redefine it here